### PR TITLE
Refs #5383: Document that aliasing dataclass fails

### PR DESCRIPTION
**Pull request by wkschwartz**
This documentation is a temporary solution. It should be altered or
removed when #5383 is fixed.

I inserted the whole example from #5383. If you think that's too much, feel free to modify. I've granted edit permissions to maintainers.

### CODE REVIEW

File: docs/source/additional_features.rst [hunk 1]
```diff
@@ -68,9 +68,38 @@ class can be used:
...
+
+    from dataclasses import dataclass
+
+    dataclass_alias = dataclass
+    dataclass_alias2 = dataclass(order=False)
+    def dataclass_wrapper(cls):
+      return dataclass(cls)
+
+    @dataclass_alias
+    class BadlyDecoratedClass1: arg1: int
```
Comment (JelleZijlstra) [inline docs/source/additional_features.rst]: I didn't realize this syntax was legal. I think it's better to put the field on a separate line.

**Comment by gvanrossum** [comment]
@wkschwartz Please respond to the review feedback. (Please use a new local commit, or several, to update your changes -- don't use `git commit --amend; git push -f`.)

File: docs/source/additional_features.rst [hunk 2]
```diff
@@ -68,9 +68,38 @@ class can be used:
...
-**Note:** Some functions in the ``dataclasses`` module, such as ``replace()`` and ``asdict()``,
+Caveats/Known Issues
+====================
+
+Some functions in the ``dataclasses`` module, such as ``replace()`` and ``asdict()``,
 have imprecise (too permissive) types. This will be fixed in future releases.
 
+Mypy does not yet recognize aliases of ``dataclasses.dataclass``, and will
+probably never recognize dynamically computed decorators. The following examples
+do **not** work:
```
Comment (ilevkivskyi) [inline docs/source/additional_features.rst]: I would actually skip the examples, the textual formulation is already clear (except maybe the dynamically computed decorators).

**Comment by wkschwartz** [comment]
@ilevkivskyi I removed one of the three examples in response to https://github.com/python/mypy/pull/5387#discussion_r206927049 in dc742c8

File: docs/source/additional_features.rst [hunk 3]
```diff
@@ -68,9 +68,43 @@ class can be used:
...
+    def dataclass_wrapper(cls):
+      return dataclass(cls)
+
+    @dataclass_alias
+    class AliasDecorated:
+      """
+      Mypy doesn't recognize this as a dataclass because it is decorated by an
+      alias of `dataclass` rather than by `dataclass` itself.
+      """
+      arg1: int
```
Comment (ilevkivskyi) [inline docs/source/additional_features.rst]: Could you please use `attr` (or similar) instead of `arg1`?

**Pull request merged**